Checkly Groups Makeover - First Draft of dev docs rewrite (#1264)
checkly/docs.checklyhq.com: site/content/docs/cli/constructs-reference.md

"{ source: 'RESPONSE_DATA', regex: '', property: '', comparison: 'CONTAINS', target: 'ping' }"
```

## `CheckGroupV2`

Use the CheckGroupV2 construct to organize your Checks into groups. This comes with the following benefits:

1. Your Checks are organized in a folder in the Checkly web UI.
2. You can trigger all Checks in a group from the web UI and via a command line trigger.
3. You can manage group-level configuration like the runtime, activated & muted-state, tags and alert channels that trickle down to all the Checks in the group.

> [!WARNING]

> > Adding a check to a group that has defined group-level alert settings means the check will _only_ alert through the group's alert channels. Make sure your group has connected alert channels, or you might miss out on important alerts!

Reviewer: ```suggestion
> If you add a check to a group with group-level alert settings ON, the check will only send alerts through the group’s alert channels. Make sure the group has at least one alert channel connected, otherwise you won’t receive any alerts for that check.
```

- `alertChannels`: An array of [AlertChannel](#alertchannel) objects to be alerted on when checks in this group fail or recover.
- `tags`: An array of tags i.e. `['product', 'api']`. Group tags trickle down to tags on the individual Checks.
- `runtimeId`: The ID of which [runtime](/docs/runtimes/specs/) to use for the Checks in the group. Use this if the checks in this group require a different runtime than your account default.
- `environmentVariables`: An array of objects defining [environment variables in the group scope](/docs/groups/variables), i.e. `[{ key: 'DEBUG', value: 'true', secret: true | locked: true }]`.
- `localSetupScript`: Any JS/TS code as a string to run before each API Check in this group.
- `localTearDownScript`: Any JS/TS code as a string to run after each API Check in this group.
- `retryStrategy`: A [RetryStrategy](#retrystrategy) object configuring [retries](/docs/alerting-and-retries/) for failed Check runs. If set, all checks in the group use the group's retry strategy. If not set, individual Check settings are used.

> - `apiCheckDefaults`: A set of [defaults for API Checks](/docs/groups/api-check-defaults/). This should not be needed. Just compose shared defaults using JS/TS.

Reviewer: what's apiCheckDefaults? given this text I'd deprecate this option entirely. Such a funny paragraph you run into :) 

```suggestion
- `apiCheckDefaults`: Deprecated. [A set of shared values for API Checks](/docs/groups/api-check-defaults/).  Instead, use JavaScript or TypeScript to define shared defaults directly in your code.
```
we'd also need to deprecate in the docs page 😅

Author: Yeah so I was waiting for Simo's input on this but maybe you can help me out here as well - How come they are deprecated? We continue to support them in the UI...

![Uploading Screenshot 2025-06-06 at 14.59.06.png…]()

Reviewer: they're not deprecated as far as I can see - I'm proposing that they do become deprecated based on the docs saying "you don't need this, just use JS/TS code to do it" - I'd check usage before doing such deprecation, but still, felt inconsistent

- `browserChecks`: A set of defaults for Browser Checks. This should not be needed. Just compose shared defaults using JS/TS.
- `runParallel`: A boolean that controls how Checks in the group are executed across locations. When true, all Checks run in parallel across all configured locations. When false, they run in round-robin mode. If not set, each Check uses its own scheduling strategy.

> - `alertEscalationPolicy`: An [AlertEscalationPolicy](#alertescalationpolicy) object configuring [alert-settings](/docs/alerting-and-retries/) for Check runs. If set, this overrides the alert settings of all checks in the group. If not set, each Check uses its own alert configuration.

Reviewer: ```suggestion
- `alertEscalationPolicy`: An [AlertEscalationPolicy](#alertescalationpolicy) object defines [alert-settings](/docs/alerting-and-retries/) for Check runs. If **set**, it overrides the alert settings of all checks in the group. If **not set**, each Check uses its own alert configuration.
```

> })

Reviewer 2: @sujaya-sys do you mean how `group` is defined in the definition of the API check below?

```
[...]

new ApiCheck('check-group-api-check-1', {
  name: 'API check #1',
  group,
  request: {
    method: 'GET',
    url: 'https://mac-demo-repo.vercel.app/api/hello',
  }
})
```

In that case, yes. That's how check groups get associated with a check, it ends up translating to:

```
[...]

new ApiCheck('check-group-api-check-1', {
  name: 'API check #1',
  group: {
      name: 'Group',
      activated: true,
      frequency: Frequency.EVERY_15M,
      locations: ['us-east-1', 'eu-west-1'],
      tags: ['api-group'],
      concurrency: 10,
      browserChecks: {
      frequency: Frequency.EVERY_30M,
      testMatch: '*.spec.js'
  },
  request: {
    method: 'GET',
    url: 'https://mac-demo-repo.vercel.app/api/hello',
  }
})
```

You can check an example used by checkly here, where the [check group is defined](https://github.com/checkly/checkly-webapp/blob/2fb37d73b7c68416fe78ec7f72e6f5735c5d3962/__checks__/external-services.check-group.ts) and then [referenced in the check](https://github.com/checkly/checkly-webapp/blob/d983e3e47426be01a9d49b43b77dbdba359c54a3/src/services/__checks__/get_nearest_regions.check.ts#L6).

> })

Author: Got it, thanks a lot for the explanation!

```

- `name` (required): A friendly name for your Check Group.
- `activated` (required): Boolean that determines whether the checks in the group are running or not. When set to true, all activated checks within the group will run. When set to false, no checks in the group will run, regardless of whether they are activated or not.

> - `muted` (required): Boolean that controls alerting behavior for the group. When set to true, no alerts will be sent for any checks in the group — even if the individual checks are not muted. When set to false, alerting follows the individual check’s muted setting.

Reviewer 2: @sujaya-sys These two were **not** required fo v1, are they gonna be required for v2?

Author: I think you were looking at an outdated version, already removed :) They won't be required in v2 after all (Slack thread for reference: https://checklyhq.slack.com/archives/C08BNCNL1L2/p1749041559116909)

- `alertEscalationPolicy`: An [AlertEscalationPolicy](#alertescalationpolicy) object configuring [alert-settings](/docs/alerting-and-retries/) for check runs. If set, this overrides the alert settings of all checks in the group. If not set, each check uses its own alert configuration.

> When adding checks to a group using `testMatch`, the CLI searches for files using the corresponding [check file](/docs/cli/using-check-test-match/#checkscheckmatch) as a base path.

> Note that you can configure two different `frequency` properties for API and Browser checks in a `CheckGroup` separately.
> The CLI follows a fallback logic using `Check->CheckGroup->Project` configurations.

## `CheckGroup` (deprecated)

As of CLI release vX.X.X, the original CheckGroup construct is deprecated and will be removed in a future version. We recommend migrating to [CheckGroupV2](#checkgroupv2), which offers more intuitive behavior and better control.

For a full overview of what’s changing and what to watch out for, check out our [migration guide]().

> In short: In the deprecated CheckGroup, the properties runParallel, locations, alertEscalationPolicy, and retryStrategy were always treated as overrides — even when not explicitly set, default values were applied and used to override the check’s individual settings.With CheckGroupV2, no value means no override. If these fields are left undefined, checks will use their own individual configuration. 

Author: I think you were looking at an outdated version of the document here, what you're suggestion is already in place :) LMK if I'm missing anything.
